onnxruntime/core/graph/ep_graph_utils.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/session/onnxruntime_c_api.h"
+#include "core/session/ort_apis.h"
+#include "core/graph/model.h"
+#include "core/graph/graph.h"
+#include "core/graph/ep_api_types.h"
+#include "core/graph/graph_utils.h"
+#include "core/framework/error_code_helper.h"
+
+#include <memory>
+
+namespace onnxruntime {
+namespace ep_graph_utils {
+OrtStatusPtr GetSubGraphAsModelFromGraph(const OrtGraph* src_graph,
```

Comment:
nit: perhaps this function should return an `onnxruntime::Status` and the caller can convert to `OrtStatus*` if necessary.